--- gen/levels/game_over_level.orig.s	2015-06-30 15:19:02.339548628 -0500
+++ gen/levels/game_over_level.s	2015-06-30 15:21:52.863971433 -0500
@@ -21,9 +21,7 @@
 .word 0x19040002
 .word 0x170C0013, __load_behavior_data, __load_behavior_data_end
 .word 0x1A0C000A, mario_bg, mario_bg_end
-.word 0x1D040000
-.word 0x1F080100, 0x1400035C
-.word 0x20040000
+.word 0x01100014, game_over_level, game_over_level_end, 0x1400020C
 .word 0x1E040000
 .word 0x03040002
 .word 0x34040000
@@ -41,9 +39,7 @@
 .word 0x19040003
 .word 0x170C0013, __load_behavior_data, __load_behavior_data_end
 .word 0x1A0C000A, mario_bg, mario_bg_end
-.word 0x1D040000
-.word 0x1F080100, 0x140003B8
-.word 0x20040000
+.word 0x01100014, game_over_level, game_over_level_end, 0x1400020C
 .word 0x1E040000
 .word 0x03040002
 .word 0x34040000
